analysis/src/vm/batcher.rs

```diff
@@ -1,15 +1,17 @@
 //! Batch compatible statements in each function of each machine
```

Comment:
A lot of these changes are just moving to using a visitor. I could potentially move this to a PR before the module system.

Reply:
Yes, please do that :)

Reply:
I ended up reverting everything after type checking, as we can just keep the same structure as before, using absolute paths instead of names.